src/index.js

```diff
@@ -1,2 +1,54 @@
 /* eslint-disable react/react-in-jsx-scope, react/jsx-filename-extension */
 /* @jsx createElement */
+function createElement(tagName, props, ...children) {
+  const element = document.createElement(tagName);
+
+  Object.entries(props || {}).forEach(([key, value]) => {
+    element[key.toLowerCase()] = value;
+  });
+
+  children.flat().forEach((child) => {
+    if (child instanceof Node) {
+      element.appendChild(child);
+      return;
+    }
+    element.appendChild(document.createTextNode(child));
+  });
+  return element;
+}
+
+function counter(count = 0) {
+  function handleClickCounter(number) {
+    counter(number + 1);
+  }
+
+  function handleClickNumber(number) {
+    counter(number);
+  }
+
+  const countElement = (
+    <div id="hello" className="greeting">
+      <p>Hello, world!</p>
+      <p>
+        <button type="button" onClick={() => handleClickCounter(count)}>
```

Comment:
```suggestion
        <button type="button" onClick={handleClickCounter}>
```

클래스로 작성했어서 오류가 발생했는데 마찬가지로 함수형에서 onClick 에 핸들러 함수를 넘겨주는 형태로 사용하는 것과 즉시 실행하는 형태로 사용하는 것에는 어떤 차이가 있을까요?

Reply:
class에서 함수형으로 변경하면서 미쳐 바꾸지 못한 코드로 인해 핸들러 함수를 사용하였는데요
의견 주신 것을 통해 불필요한 코드를 개선한 결과
즉시 실행하는 형태로 변경하여 클릭 시 counter 함수를 재호출하는 형태로 변경하였습니다.